### PR TITLE
Add support for Node.js 12

### PR DESCRIPTION
**Pull request by ahmetcetin**
setting node version the same as in loopback-datasource-juggler to support new versions of node.

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-context) 👈

- [ ] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](https://loopback.io/doc/en/contrib/style-guide-es6.html)
- [ ] Commit messages are following our [guidelines](https://loopback.io/doc/en/contrib/git-commit-messages.html)

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

File: package.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "version": "3.4.0",
   "description": "Current context for LoopBack applications, based on cls-hooked",
   "engines": {
-    "node": "^8.2.1 || ^10.14"
+    "node": ">=8"
```
Comment (bajtos) [inline package.json]: @ahmetcetin please note that we are intentionally limiting the Node.js versions supported by this module. The implementation depends on Node.js functionality that used to change between major versions of Node, we need to test the module against each new major version before we can claim it as being supported.

I am proposing the following version spec:

```suggestion
    "node": "^8.2.1 || ^10.14 || ^12.15"
```

**Review by bajtos — CHANGES_REQUESTED** [review]
Thank you for the pull request. I am fine to expand the range of supported Node.js versions, but we need to change the way how it's done, see my comment below.

**Comment by bajtos** [comment]
@slnode ok to test

**Comment by bajtos** [comment]
@ahmetcetin one more thing: please update Travis CI configuration to run the tests on Node.js 12 version, so that we can verify that loopback-context actually does support it.

See here:

https://github.com/strongloop/loopback-context/blob/a575d934a7c0a9895e4c9baa714da792b487f4f2/.travis.yml#L3-L5

**Comment by ahmetcetin** [comment]
All requested changes done.

**Comment by nabdelgadir** [comment]
Thanks @ahmetcetin for this PR! Before we can land it, can you squash your commits into one commit (see [here](https://loopback.io/doc/en/lb4/submitting_a_pr.html#10-final-rebase-and-squashing-of-commits) if you need help)? Also, make sure to use new lines in your commit message so no lines exceed 72 characters. Here's a suggestion:

```
chore: add support for Node.js 12

> Setting the Node.js version to be the same as in 
> loopback-datasource-juggler to support new versions of Node.js
```

**Comment by bajtos** [comment]
IMO, adding support for a new Node.js version is not a `chore`, it's a `feat`ure. 

Having said that, we are not following conventional commit conventions in this repository, I think the following would be the best commit summary:

```
Add support for Node.js 12
```

**Review by bajtos — APPROVED** [review]

**Comment by bajtos** [comment]
Landed, thank you @ahmetcetin for the contribution!

**Pull request merged**